upf/upf.c

```diff
@@ -109,7 +106,9 @@ upf_init_nat_addresses (upf_nat_pool_t * np, ip4_address_t start_addr,
 {
   u32 i = 0;
 
-  vec_reset_length (np->addresses);
+  vec_validate (np->addresses,
+		clib_net_to_host_u32 (end_addr.as_u32) -
+		clib_net_to_host_u32 (start_addr.as_u32) + 1);
```

Comment:
I did suggest this change, but that does mean that you should turn off mind and copy what I proposed without thinking.

Simply replacing the` vec_reset_length` with `vec_validate` is neither sufficient nor correct.

Also, the commit should be folded into the NAT implementation commit.